stan/math/prim/prob/loglogistic_lpdf.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef STAN_MATH_PRIM_PROB_LOGLOGISTIC_LPDF_HPP
+#define STAN_MATH_PRIM_PROB_LOGLOGISTIC_LPDF_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/as_column_vector_or_scalar.hpp>
+#include <stan/math/prim/fun/as_array_or_scalar.hpp>
+#include <stan/math/prim/fun/as_value_column_array_or_scalar.hpp>
+#include <stan/math/prim/fun/exp.hpp>
+#include <stan/math/prim/fun/log.hpp>
+#include <stan/math/prim/fun/max_size.hpp>
+#include <stan/math/prim/fun/size.hpp>
+#include <stan/math/prim/fun/size_zero.hpp>
+#include <stan/math/prim/fun/to_ref.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+#include <stan/math/prim/functor/operands_and_partials.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/** \ingroup prob_dists
+ * The log of the loglogistic density for the specified scalar(s)
+ * given the specified scales(s) and shape(s). y, alpha, or beta
+ * can each be either a scalar or a vector. Any vector inputs
+ * must be the same length.
+ *
+ * <p>The result log probability is defined to be the sum of the
+ * log probabilities for each observation/scale/shape triple.
+ *
+ * @tparam T_y type of scalar.
+ * @tparam T_scale type of scale parameter.
+ * @tparam T_shape type of shape parameter.
+ * @param y (Sequence of) scalar(s).
+ * @param alpha (Sequence of) scale parameter(s)
+ * for the loglogistic distribution.
+ * @param beta (Sequence of) shape parameter(s) for the
+ * loglogistic distribution.
+ * @return The log of the product of the densities.
+ * @throw std::domain_error if any of the inputs are not positive and finite.
+ */
+template <bool propto, typename T_y, typename T_scale, typename T_shape,
+          require_all_not_nonscalar_prim_or_rev_kernel_expression_t<
+              T_y, T_scale, T_shape>* = nullptr>
+return_type_t<T_y, T_scale, T_shape> loglogistic_lpdf(const T_y& y,
+                                                      const T_scale& alpha,
+                                                      const T_shape& beta) {
+  using T_partials_return = partials_return_t<T_y, T_scale, T_shape>;
+  using T_y_ref = ref_type_if_t<!is_constant<T_y>::value, T_y>;
+  using T_scale_ref = ref_type_if_t<!is_constant<T_scale>::value, T_scale>;
+  using T_shape_ref = ref_type_if_t<!is_constant<T_shape>::value, T_shape>;
+  using std::pow;
+  static const char* function = "loglogistic_lpdf";
+  check_consistent_sizes(function, "Random variable", y, "Scale parameter",
+                         alpha, "Shape parameter", beta);
+
+  T_y_ref y_ref = y;
+  T_scale_ref alpha_ref = alpha;
+  T_shape_ref beta_ref = beta;
+
+  decltype(auto) y_val = to_ref(as_value_column_array_or_scalar(y_ref));
+  decltype(auto) alpha_val = to_ref(as_value_column_array_or_scalar(alpha_ref));
+  decltype(auto) beta_val = to_ref(as_value_column_array_or_scalar(beta_ref));
+
+  check_positive_finite(function, "Random variable", y_val);
```

Comment:
@t4c1 should I change line 65 to:
`check_not_nan(function, "Random variable", y_val);`
`check_positive(function, "Random variable", y_val);`?
Theoretically, `y_val` can be infinite. And I haven't seen checks for finite `y_val` in other similar cases, such as `normal_lpdf` and `gamma_lpdf`. But there is a check for finite `y_val` in the `logistic_lpdf`.

Reply:
I am not sure when do isfinite checks make sense and when not. Even if you decide to replace `check_positive_finite` with `check_positive`, there is no need for `check_not_nan`. A NaN value will fail any check.

Reply:
OK, thanks!